sympy/polys/sqfreetools.py

```diff
@@ -70,36 +86,86 @@ def dmp_sqf_p(f, u, K):
     """
     if dmp_zero_p(f, u):
         return True
-    else:
-        return not dmp_degree(dmp_gcd(f, dmp_diff(f, 1, u, K), u, K), u)
+
+    for i in range(u+1):
+
+        fp = dmp_diff_in(f, 1, i, u, K)
+
+        if dmp_zero_p(fp, u):
+            continue
+
+        gcd = dmp_gcd(f, fp, u, K)
+
+        if dmp_degree_in(gcd, i, u) != 0:
+            return False
+
+    return True
 
 
 def dup_sqf_norm(f, K):
-    """
-    Square-free norm of ``f`` in ``K[x]``, useful over algebraic domains.
+    r"""
+    Find a shift of `f` in `K[x]` that has square-free norm.
+
+    The domain `K` must be an algebraic number field `k(a)` (see :ref:`QQ(a)`).
 
-    Returns ``s``, ``f``, ``r``, such that ``g(x) = f(x-sa)`` and ``r(x) = Norm(g(x))``
-    is a square-free polynomial over K, where ``a`` is the algebraic extension of ``K``.
+    Returns `(s,g,r)`, such that `g(x)=f(x-sa)`, `r(x)=\text{Norm}(g(x))` and
+    `r` is a square-free polynomial over `k`.
```

Comment:
Don't all the single backticks need to be double? (More below)

Reply:
It depends whether you want LaTeX or code. Single backticks gives LaTeX like $r(x)=\text{Norm}(g(x))$ and double backticks gives code like `r(x) = Norm(g(x))`. I'm not sure what should be preferred in these docstrings.

Reply:
Oh, never realized we had `default_role = 'math'` in the docs. Sure, LaTeX seems fine.